src/traces/image/plot.js

```diff
@@ -181,13 +175,23 @@ module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
             }
         })
         .then(function() {
-            var href, canvas;
+            var href, canvas, localStyle;
+            localStyle = initialStyle;
             if(trace._hasZ) {
                 canvas = drawMagnifiedPixelsOnCanvas(function(i, j) {return z[j][i];});
                 href = canvas.toDataURL('image/png');
             } else if(trace._hasSource) {
                 if(fastImage) {
                     href = trace.source;
+                    // Flip the SVG image as needed (around the proper center location)
+                    var axisScale = [
+                        (xa.range[0] < xa.range[1]) ? 1 : -1,
```

Comment:
If you console.log xa.range here what do you get from that failing test?

Reply:
```
LOG: [0, 512]
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct hover info regardless of axis directions 0,512,0,512 FAILED
LOG: [0, 512]
-> this test passes
LOG: [512, 0]
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct 
LOG: [512, 0]
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct hover info regardless of axis directions 512,0,512,0 FAILED
```

Reply:
No. With autorange?

Reply:
I also replaced autorange '"auto"` with `true`. The "auto" was why it did work for the y-axis earlier.

```
LOG: Object{xa.range: [-0.5, 511.5], ya.range: [-0.5, 511.5]}
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct hover info regardless of axis directions true,true FAILED
LOG: Object{xa.range: [-0.5, 511.5], ya.range: [511.5, -0.5]}
LOG: Object{xa.range: [511.5, -0.5], ya.range: [-0.5, 511.5]}
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct hover info regardless of axis directions reversed,true FAILED
LOG: Object{xa.range: [511.5, -0.5], ya.range: [511.5, -0.5]}
Chrome 87.0.4280.66 (Windows 10) image hover: for `image_astronaut_source` defined by source should show correct hover info regardless of axis directions reversed,reversed FAILED
```